File: src/Entity/Produit.php
<?php

namespace App\Entity;

use App\Repository\ProduitRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;
/**
 * @ORM\Entity(repositoryClass=ProduitRepository::class)
 */
class Produit
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     */
    private $id;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank(message="type produit is required")
     */
    private $type_produit;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank(message="prix is required")
     */
    public $prix_produit;

    /**
     * @ORM\Column(type="date")
     ** @Assert\NotBlank(message="date expo is required")
     */
    private $date_expo;

    /**
     * @ORM\Column(type="date")
     * @Assert\NotBlank(message="date fin is required")
     */
    private $date_fin;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank(message="disponibilité is required")
     */
    private $disponibilite;

    /**
     * @ORM\Column(type="string", length=255)
     *@Assert\NotBlank(message="type action is required")
     */
    private $type_action
    ;

    /**
     * @var string
     *
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank(message="image is required")
     */
    private $imgproduit;

    /**
     * @ORM\OneToMany(targetEntity=LigneCommande::class, mappedBy="Produit")
     */
    private $ligneCommandes;

    public function __construct()
    {
        $this->ligneCommandes = new ArrayCollection();
    }



    public function getId(): ?int
    {
        return $this->id;
    }

    public function getTypeProduit(): ?string
    {
        return $this->type_produit;
    }

    public function setTypeProduit(string $type_produit): self
    {
        $this->type_produit = $type_produit;

        return $this;
    }

    public function getPrixProduit(): ?string
    {
        return $this->prix_produit;
    }

    public function setPrixProduit(string $prix_produit): self
    {
        $this->prix_produit = $prix_produit;

        return $this;
    }

    public function getImgproduit(): ?string
    {
        return $this->imgproduit;
    }

    public function setImgproduit(string $imgproduit): self
    {
        $this->imgproduit = $imgproduit;

        return $this;
    }



    public function getDateExpo(): ?\DateTimeInterface
    {
        return $this->date_expo;
    }

    public function setDateExpo(\DateTimeInterface $date_expo): self
    {
        $this->date_expo = $date_expo;

        return $this;
    }

    public function getDateFin(): ?\DateTimeInterface
    {
        return $this->date_fin;
    }

    public function setDateFin(\DateTimeInterface $date_fin): self
    {
        $this->date_fin = $date_fin;

        return $this;
    }

    public function getDisponibilite(): ?string
    {
        return $this->disponibilite;
    }

    public function setDisponibilite(string $disponibilite): self
    {
        $this->disponibilite = $disponibilite;

        return $this;
    }

    public function getTypeAction(): ?string
    {
        return $this->type_action;
    }

    public function setTypeAction(string $type_action): self
    {
        $this->type_action = $type_action;

        return $this;
    }

    public function __toString()
    {
        return $this->getTypeProduit();
    }

    /**
     * @return Collection|LigneCommande[]
     */
    public function getLigneCommandes(): Collection
    {
        return $this->ligneCommandes;
    }

    public function addLigneCommande(LigneCommande $ligneCommande): self
    {
        if (!$this->ligneCommandes->contains($ligneCommande)) {
            $this->ligneCommandes[] = $ligneCommande;
            $ligneCommande->setProduit($this);
        }

        return $this;
    }

    public function removeLigneCommande(LigneCommande $ligneCommande): self
    {
        if ($this->ligneCommandes->removeElement($ligneCommande)) {
            // set the owning side to null (unless already changed)
            if ($ligneCommande->getProduit() === $this) {
                $ligneCommande->setProduit(null);
            }
        }

        return $this;
    }




}

File: src/Entities/Produit.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Entities;

import java.util.Date;

/**
 *
 * @author Administrateur
 */
public class Produit {
    private int id ;
    private String type_produit;
    private int prix_produit;
    private Date date_expo;
     private Date date_fin;
        private String disponibilite;
     private String type_action;
     private String imgproduit;

    public Produit() {
    }

    public Produit(int id, String type_produit, int prix_produit, Date date_expo, Date date_fin, String disponibilite, String type_action, String imgproduit) {
        this.id = id;
        this.type_produit = type_produit;
        this.prix_produit = prix_produit;
        this.date_expo = date_expo;
        this.date_fin = date_fin;
        this.disponibilite = disponibilite;
        this.type_action = type_action;
        this.imgproduit = imgproduit;
    }

    public Produit(int id) {
        this.id = id;
    }

    public Produit(String type_produit, int prix_produit, Date date_expo, Date date_fin, String disponibilite, String type_action, String imgproduit) {
        this.type_produit = type_produit;
        this.prix_produit = prix_produit;
        this.date_expo = date_expo;
        this.date_fin = date_fin;
        this.disponibilite = disponibilite;
        this.type_action = type_action;
        this.imgproduit = imgproduit;
    }

    @Override
    public String toString() {
        return "Produit{" + "id=" + id + ", type_produit=" + type_produit + ", prix_produit=" + prix_produit + ", date_expo=" + date_expo + ", date_fin=" + date_fin + ", disponibilite=" + disponibilite + ", type_action=" + type_action + ", imgproduit=" + imgproduit + '}';
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getType_produit() {
        return type_produit;
    }

    public void setType_produit(String type_produit) {
        this.type_produit = type_produit;
    }

    public int getPrix_produit() {
        return prix_produit;
    }

    public void setPrix_produit(int prix_produit) {
        this.prix_produit = prix_produit;
    }

    

    public Date getDate_expo() {
        return date_expo;
    }

    public void setDate_expo(Date date_expo) {
        this.date_expo = date_expo;
    }

    public Date getDate_fin() {
        return date_fin;
    }

    public void setDate_fin(Date date_fin) {
        this.date_fin = date_fin;
    }

    public String getDisponibilite() {
        return disponibilite;
    }

    public void setDisponibilite(String disponibilite) {
        this.disponibilite = disponibilite;
    }

    public String getType_action() {
        return type_action;
    }

    public void setType_action(String type_action) {
        this.type_action = type_action;
    }

    public String getImgproduit() {
        return imgproduit;
    }

    public void setImgproduit(String imgproduit) {
        this.imgproduit = imgproduit;
    }

                    
    
    
    
}
